test/system/src/test/java/io/pravega/test/system/RetentionTest.java

```diff
@@ -167,15 +164,14 @@ public void retentionTest() throws Exception {
 
         //try reading the event that was written earlier.
         //expectation is it should have been truncated and we should find stream to be empty
-        try {
-            String readEvent = reader.readNextEvent(6000).getEvent();
-            log.debug("Reading event: {} ", readEvent);
-            assertEquals(null, readEvent);
-        }  catch (ReinitializationRequiredException e) {
-            log.error("Unexpected request to reinitialize {}", e);
-            Assert.fail("Unexpected request to reinitialize.Test failed.");
-        }
+        assertThrows(TruncatedDataException.class, () -> reader.readNextEvent(6000));
+
+        //verify reader functionality is unaffected post truncation
+        String event = "newEvent";
+        writer.writeEvent(event);
+        log.info("Writing event: {}", event);
+        Assert.assertEquals(event, reader.readNextEvent(6000).getEvent());
 
-       log.debug("The stream is already truncated.Simple retention test passed.");
+        log.debug("The stream is already truncated.Simple retention test passed.");
```

Comment:
Remove additional leading space.

Reply:
I have actually corrected the wrong indentation here.